### PR TITLE
Feature deal and place editor. Feature settings page

### PR DESCRIPTION
**Pull request by artur-masalcev**

### CODE REVIEW

File: Wasted/Pages/Place/DealEditorPage.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            BindingContext = this;
+        }
+
+        /// <summary>
+        /// Checks whether all user fields are not empty
+        /// </summary>
+        /// <returns>'false' if any of fields is empty. 'true' if all fields are not empty</returns>
+        private bool IsDataValid()
+        {
+            return !StringUtils.AnyNullOrEmpty(NewTitle, NewCurrentCost,
```
Comment (Mantas330BJ) [inline Wasted/Pages/Place/DealEditorPage.xaml.cs:50]: We already have a similar function for checking if data is empty I think in one of exception utils classes

File: Wasted/Pages/Place/DealEditorPage.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+        /// <summary>
+        /// Publishes updated deal data to the server.
+        /// Also checks whether the data is valid, if it is not - alert is displayed
+        /// </summary>
+        private void SaveChangesClicked(object sender, EventArgs e)
+        {
+            if (!IsDataValid())
+            {
+                DisplayAlert("", "Please fill all fields", "OK");
```
Comment (Mantas330BJ) [inline Wasted/Pages/Place/DealEditorPage.xaml.cs:101]: We already have a similar method for handling empty data in one of exception utils classes

File: Wasted/Pages/Place/FoodPlacePage.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        
+        private void DealsCollectionViewSelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            Deal selectedDeal = e.CurrentSelection.First() as Deal;
+            Navigation.PushAsync(new DealEditorPage(selectedDeal));
+        }
+        
+        private void BackClicked(object sender, EventArgs e)
+        {
+            Navigation.PopAsync(true);
```
Comment (Mantas330BJ) [inline Wasted/Pages/Place/FoodPlacePage.xaml.cs:54]: If I remember correctly, true is default value and can be omitted

File: Wasted/Pages/Place/PageManager.cs [hunk 4]
```diff
@@ -7,12 +7,12 @@ public static class PageManager
...
         public static void PushClientPage(ContentPage page)
         {
-            page.Navigation.PushAsync(new MainPage());
+            page.Navigation.PushAsync(new Client.MainPage());
         }
 
         public static void PushPlacePage(ContentPage page)
         {
-            page.Navigation.PushAsync(new PlaceMainPage());
+            page.Navigation.PushAsync(new MainPage());
```
Comment (Mantas330BJ) [inline Wasted/Pages/Place/PageManager.cs:15]: What was the purpose of renaming this class?

File: Wasted/WastedAPI/DataProvider.cs [hunk 5]
```diff
@@ -58,22 +57,25 @@ private static void DeleteBusinessObject(int id, string linkEnd)
         }
 
         /// <summary>
-        /// Converts object to json StringContent.
+        /// Converts object to json StringContent. Serializing content once is not enough
```
Comment (Mantas330BJ) [inline Wasted/WastedAPI/DataProvider.cs:60]: I think you did not want to change this

File: Wasted/Pages/Place/PageManager.cs (hunk 4 above)
Comment (artur-masalcev) [reply Wasted/Pages/Place/PageManager.cs:15]: Just to fit the style of Client type user, where we have "MainPage" and not "ClientMainPage". Also the old name can be confusing when there are such files as "FoodPlacePage".

**Pull request merged**